java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -355,10 +359,16 @@ public static Table readTableFrom(InputStream in) throws IOException {
       din = new DataInputStream(in);
     }
 
-    int num = din.readInt();
-    if (num != SER_FORMAT_MAGIC_NUMBER) {
-      throw new IllegalStateException("THIS DOES NOT LOOK LIKE CUDF SERIALIZED DATA. " +
-          "Expected magic number " + SER_FORMAT_MAGIC_NUMBER + " Found " + num);
+    try {
+      int num = din.readInt();
+      if (num != SER_FORMAT_MAGIC_NUMBER) {
+        throw new IllegalStateException("THIS DOES NOT LOOK LIKE CUDF SERIALIZED DATA. " +
+            "Expected magic number " + SER_FORMAT_MAGIC_NUMBER + " Found " + num);
+      }
+    } catch (EOFException e) {
+      // If we get an EOF at the very beginning don't treat it as an error becasue we may
```

Comment:
nit, becasue->because.

Reply:
Thanks, I spell checked the entire file.